src/server/scripts/Outland/BlackTemple/boss_illidan.cpp

```diff
@@ -50,2180 +37,2571 @@ enum Say
+        void MovementInform(uint32 type, uint32 pointId) override
         {
-            if (victim->GetTypeId() != TYPEID_PLAYER)
+            if (type != POINT_MOTION_TYPE && type != EFFECT_MOTION_TYPE)
                 return;
 
-            Talk(SAY_ILLIDAN_KILL);
+            switch (pointId)
+            {
+                case POINT_THROW_GLAIVE:
+                    DoPlaySoundToSet(me, ILLIDAN_WARGLAIVE_SOUND_ID);
+                    events.ScheduleEvent(EVENT_THROW_WARGLAIVE, Seconds(2), GROUP_PHASE_ALL);
+                    events.ScheduleEvent(EVENT_FACE_MIDDLE, Milliseconds(1), GROUP_PHASE_ALL);
+                    break;
+                case POINT_RANDOM_PILLAR:
+                    ScheduleEvents(GROUP_PHASE_2, GROUP_PHASE_2);
+                    break;
+                case POINT_ILLIDAN_MIDDLE:
+                {
+                    std::list<Creature*> triggers;
+                    GetCreatureListWithEntryInGrid(triggers, me, NPC_BLADE_OF_AZZINOTH, 150.0f);
+                    for (Creature* trigger : triggers)
+                        trigger->CastSpell(trigger, SPELL_GLAIVE_RETURNS, true);
+
+                    events.ScheduleEvent(EVENT_GLAIVE_EMOTE, Seconds(3), GROUP_PHASE_ALL);
+                    break;
+                }
+                default:
+                    break;
+            }
         }
 
-        void DamageTaken(Unit* done_by, uint32 &damage) override
+        void EnterEvadeModeIfNeeded()
         {
-            if (damage >= me->GetHealth() && done_by != me)
-                damage = 0;
-            if (done_by->GetGUID() == MaievGUID)
-                done_by->AddThreat(me, -(3*(float)damage)/4); // do not let maiev tank him
+            Map::PlayerList const &players = me->GetMap()->GetPlayers();
+            for (Map::PlayerList::const_iterator i = players.begin(); i != players.end(); ++i)
+                if (Player* player = i->GetSource())
+                    if (player->IsAlive() && !player->IsGameMaster() && CheckBoundary(player))
+                        return;
+
+            EnterEvadeMode(EVADE_REASON_NO_HOSTILES);
         }
 
-        void SpellHit(Unit* /*caster*/, const SpellInfo* spell) override
+        void SummonMinions()
         {
-            if (spell->Id == SPELL_GLAIVE_RETURNS) // Re-equip our warglaives!
+            uint8 needSummon = MAX_MINIONS_NUMBER - _minionsCount;
+            for (uint8 i = 0; i < needSummon; ++i)
             {
-                if (!me->GetUInt32Value(UNIT_VIRTUAL_ITEM_SLOT_ID))
-                    SetEquipmentSlots(false, EQUIP_ID_MAIN_HAND, EQUIP_UNEQUIP, EQUIP_NO_CHANGE);
-                else
-                    SetEquipmentSlots(false, EQUIP_UNEQUIP, EQUIP_ID_OFF_HAND, EQUIP_NO_CHANGE);
-                me->SetSheath(SHEATH_STATE_MELEE);
+                _minionsCount++;
+                me->SummonCreature(NPC_ILLIDARI_ELITE, MinionsSpawnPositions[i], TEMPSUMMON_CORPSE_TIMED_DESPAWN, 6000);
```

Comment:
i did it by hand, because in retail illidan summon groups each 30 secs. But when akama dont kill this first group, illidan just summon creatures to complete 10 creatures again.
If i do it with summon_groups i will always create 10 creatures.
btw in TC, akama kill creatures fast, so... maybe can be changed or akama damage fixed